programs/clearing_house/src/math/auction.rs

```diff
@@ -9,26 +9,42 @@ use crate::state::user::Order;
 use solana_program::msg;
 use std::cmp::min;
 
-pub fn calculate_auction_end_price(
+pub fn calculate_auction_prices(
     oracle_price: &OraclePriceData,
     direction: PositionDirection,
-) -> ClearingHouseResult<u64> {
+    limit_price: u64,
+) -> ClearingHouseResult<(u64, u64)> {
+    if limit_price > 0 {
+        let auction_start_price = match direction {
+            PositionDirection::Long => limit_price
+                .checked_sub(limit_price / 100) // 1% improvement
+                .ok_or_else(math_error!())?,
+            PositionDirection::Short => limit_price
+                .checked_add(limit_price / 100)
+                .ok_or_else(math_error!())?,
+        };
+
+        return Ok((auction_start_price, limit_price));
+    }
+
     let numerator = match direction {
         PositionDirection::Long => {
             BID_ASK_SPREAD_PRECISION + BID_ASK_SPREAD_PRECISION / 100 // 1%
         }
         PositionDirection::Short => BID_ASK_SPREAD_PRECISION - BID_ASK_SPREAD_PRECISION / 100,
     };
 
-    cast_to_u64(
+    let auction_end_price = cast_to_u64(
         oracle_price
             .price
             .unsigned_abs()
             .checked_mul(numerator)
             .ok_or_else(math_error!())?
             .checked_div(BID_ASK_SPREAD_PRECISION)
             .ok_or_else(math_error!())?,
-    )
+    )?;
+
+    Ok((oracle_price.price.cast::<u64>()?, auction_end_price))
```

Comment:
dont really like how this would be -1% -> +1%, if limit price = order price
(with the default 10 slot auction time that .2% per auction tick which is pretty high)

better default might be:
pick a lower constant: below is .1% (could be .2% or .5%)
if limit price is worse than oracle:
 start = oracle
 end = better of: oracle improved .1% and limit price
 
 if limit is better than oracle:
 start = better of oracle improved .1% and limit improved by .1%
 end = limit